cirq-aqt/cirq_aqt/aqt_target_gateset.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Target gateset for ion trap device with mutually linked qubits placed on a line.
+"""
+
+from typing import List
+
+import numpy as np
+
+import cirq
+from cirq.protocols.decompose_protocol import DecomposeResult
+
+
+class AQTTargetGateset(cirq.TwoQubitCompilationTargetGateset):
+    """Target gateset accepting XXPowGate + X/Y/Z/PhX single qubit rotations + measurement gates.
+
+    By default, `cirq_aqt.AQTTargetGateset` will accept and compile unknown
+    gates to the following universal target gateset:
+
+    - `cirq.XXPowGate`: The two qubit entangling gate.
+    - `cirq.XPowGate`, `cirq.YPowGate`, `cirq.ZPowGate`,
+      `cirq.PhasedXPowGate`: Single qubit rotations.
+    - `cirq.MeasurementGate`: Measurements.
+    """
+
+    def __init__(self):
+        super().__init__(
+            cirq.XXPowGate,
+            cirq.MeasurementGate,
+            cirq.XPowGate,
+            cirq.YPowGate,
+            cirq.ZPowGate,
+            cirq.PhasedXPowGate,
+            unroll_circuit_op=False,
+        )
+
+    def _decompose_single_qubit_operation(self, op: 'cirq.Operation', _: int) -> DecomposeResult:
+        # unwrap tagged and circuit operations to get the actual operation
+        opu = op.untagged
+        opu = (
+            next(opu.circuit.all_operations())
```

Comment:
Optional:  do we care if the operation inside the CircuitOperation is tagged?  We could add an untagged here to handle it, just in case.

Reply:
Done in d7cf04cf741783b3f962ed76a4726df68c702b81